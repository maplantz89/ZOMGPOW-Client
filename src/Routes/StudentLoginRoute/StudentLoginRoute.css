.student-login-form{
  display: flex;
  flex-direction: column;
  padding: 15px;
  border-radius: 15px;
  border: solid 1px darkgray;
  margin: 40px auto;
  width: 80%;
  max-width: 300px;
  align-items: center;
}

.student-login-form h2 {
  font-size: 27px;
  font-family: 'Lexend Deca', sans-serif;
}

.student-login-form label {
  font-size: 20px;
  font-family: 'Lexend Deca', sans-serif;
}

.student-login-form input{
  border-radius: 5px;
  font-family: 'Montserrat', sans-serif;
  font-size: 15px;
  border: solid 1px darkgray;
  margin: 0px 10px;
}

.student-login-form p {
  margin: 0;
}

.student-login-form button{
  color: white;
  background: #0ABAD4;
  padding: 5px 15px 8px 15px;
  border-radius: 15px;
  font-size: 16px;
  text-align: center;
  font-family: 'Lexend Deca', sans-serif;
  margin-bottom: 10px;
}

.student-login-form button:hover {
  background: black;
  cursor: pointer;
}

.student-help{
  justify-content: center;
  color: #8B00B5;
  font-size: 15px;
}
