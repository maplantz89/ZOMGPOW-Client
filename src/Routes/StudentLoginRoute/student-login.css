.student-login{
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  justify-content: center;
}

.student-login-form {
  padding: 5%;
  background-color: rgb(214, 204, 204);
  border-radius: 7%;
  border: rgb(117, 115, 115) 3px solid;
  margin: 5% 40%;
  min-height: 20vh;
  
}

.student-username{
  border-radius: 2%;
  padding: 2% 0%;
  width: 80%;
  margin: 8% 0%;
}