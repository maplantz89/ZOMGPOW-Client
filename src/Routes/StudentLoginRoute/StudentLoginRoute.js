import React from 'react'
import './StudentLoginRoute.css'
import StudentAuthApiService from '../../Services/student-auth-api-service'
import StudentContext from '../../Contexts/StudentContext';


class StudentLoginRoute extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }
  static contextType = StudentContext

  state = {
    error: null,
  }

  onLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }
  
  
  
  handleSubmit = e => {
    e.preventDefault();
    const {username} = e.target
    this.setState({ error: null })

    StudentAuthApiService.postLogin({
      username: username.value,     
    })
      .then(res => {
        username.value = ''       
        this.context.processLogin(res.authToken)
        this.onLoginSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }


  render() {
    const {error} = this.state;

    return (
      <div className="student-login">
        <form className="student-login-form" onSubmit={this.handleSubmit}>
          <div role='alert'>
          {error && <p>{error}</p>}
          </div>
          <h2>Student Login</h2>
          <label>Username:</label><br></br>
          <input className= "student-username" name="student-username" required></input><br></br>
          <button type="submit">Submit</button>
          <div className='student-help'>
            <p>Forgot your username?</p>
            <p>Ask your teacher for help.</p>
          </div>
        </form>
   
      </div>
    );
  }
  
}

export default StudentLoginRoute;