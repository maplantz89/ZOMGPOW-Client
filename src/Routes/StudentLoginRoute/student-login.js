import React from 'react';
import './student-login.css';


class StudentLogin extends React.Component {

  handleSubmit = e => {
    e.preventDefault();
    const {username} = e.target
  }


  render() {


    return (
      <div className="student-login">
        <form className="student-login-form" onSubmit={this.handleSubmit}>
        <label>Username:</label><br></br>
          <input className= "student-username" name="student-username" required></input><br></br>
          <button type="submit">Submit</button>
        </form>
   
      </div>
    );
  }
  
}

export default StudentLogin;