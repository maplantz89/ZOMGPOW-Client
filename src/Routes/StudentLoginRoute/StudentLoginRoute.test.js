import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StudentLoginRoute from './StudentLoginRoute';

describe(`StudentLoginRoute component`, () => {
  it('renders a .StudentLoginRoute by default', () => {
    const wrapper = shallow(<StudentLoginRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})