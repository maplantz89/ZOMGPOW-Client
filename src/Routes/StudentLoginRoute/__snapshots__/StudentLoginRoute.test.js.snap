// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentLoginRoute component renders a .StudentLoginRoute by default 1`] = `
<div
  className="student-login"
>
  <form
    className="student-login-form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label>
      Username:
    </label>
    <br />
    <input
      className="student-username"
      name="student-username"
      required={true}
    />
    <br />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
